workflow/serializers.py

```diff
@@ -160,6 +196,40 @@ class Parameter(ModelSerializer):
         required=False
     )
 
+    def update(self, instance, validated_data):
+        raise_errors_on_nested_writes('update', self, validated_data)
+        info = model_meta.get_field_info(instance)
+
+        # Simply set each attribute on the instance, and then save it.
+        # Note that unlike `.create()` we don't need to treat many-to-many
+        # relationships as being a special case. During updates we already
+        # have an instance pk for the relationships to be associated with.
+        for attr, value in validated_data.items():
+
+            # THIS NEEDS TESTING
+            if attr in info.relations and info.relations[attr].has_through_model:
+                for related_instance in value:
+                    through_class = info.relations[attr].model_field.rel.through
+                    through_instance = through_class()
+                    existing_through_instances = through_class.objects.filter(**{
```

Comment:
a description in the comments of what this is doing would be immensely helpful also!